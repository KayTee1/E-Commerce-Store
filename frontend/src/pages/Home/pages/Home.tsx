import { Link } from "react-router-dom";

const Home = () => {
  
  return (
    <div className="flex flex-col items-center justify-center py-10">
      <h2 className="text-3xl font-bold mb-4">Welcome to React Store</h2>
      <p className="text-lg text-gray-600 mb-8">
        Your one-stop shop for all your needs.
      </p>
      
      <Link
        to="/collections"
        className="bg-blue-500 hover:bg-blue-600 text-white py-2 px-4 rounded"
      >
        Browse Products
      </Link>
    </div>
  );
};

export default Home;
