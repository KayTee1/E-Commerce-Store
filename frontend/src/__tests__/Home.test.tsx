import { describe, it, expect } from "vitest";
import { MemoryRouter } from "react-router-dom";
import Home from "../pages/Home/pages/Home";
import { createRenderer } from "react-test-renderer/shallow";

const renderer = createRenderer();

const defaultComponent = (
  <MemoryRouter initialEntries={["/"]}>
    <Home />
  </MemoryRouter>
);

describe("On the Home page", async () => {
  it("should render and match the snapshot", () => {
    renderer.render(defaultComponent);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it("should render the title", () => {
    renderer.render(defaultComponent);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toContain("Home");
  })
});
