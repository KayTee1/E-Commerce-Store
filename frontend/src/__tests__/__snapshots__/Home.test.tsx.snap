// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkout Page > Renders header 1`] = `
<Router
  location={
    {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    {
      "action": "POP",
      "createHref": [Function],
      "createURL": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "index": 0,
      "listen": [Function],
      "location": {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <CartProvider>
    <Home />
  </CartProvider>
</Router>
`;
